### PR TITLE
Update CONTRIBUTING.md to address #356

### PR DESCRIPTION
**Pull request by jd-foster**
See issue #356:
>  1. the How to Contribute to an Open Source Project on GitHub link leads to a 404 and should be updated to link to https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github
>  1. the [repo] link reference currently points to a placeholder URL (https://example.com/FIXME) and should be updated to point to the source repository for the lesson (i.e. this GitHub repository).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -101,20 +101,20 @@ community listed at <https://carpentries.org/connect/> including via social
 media, slack, newsletters, and email lists. You can also [reach us by
 email][contact].
 
-[repo]: https://example.com/FIXME
+[repo]: https://github.com/datacarpentry/sql-ecology-lesson/
 [contact]: mailto:team@carpentries.org
 [cp-site]: https://carpentries.org/
-[dc-issues]: https://github.com/issues?q=user%3Adatacarpentry
+[dc-issues]: https://github.com/issues?q=user%3Adatacarpentry/
```
Comment (adamancer) [inline CONTRIBUTING.md:107]: The trailing slash breaks this link

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -101,20 +101,20 @@ community listed at <https://carpentries.org/connect/> including via social
...
+[github]: https://github.com/
 [github-flow]: https://guides.github.com/introduction/flow/
-[github-join]: https://github.com/join
-[how-contribute]: https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github
+[github-join]: https://github.com/join/
+[how-contribute]: https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github/
 [issues]: https://carpentries.org/help-wanted-issues/
-[lc-issues]: https://github.com/issues?q=user%3ALibraryCarpentry
-[swc-issues]: https://github.com/issues?q=user%3Aswcarpentry
+[lc-issues]: https://github.com/issues?q=user%3ALibraryCarpentry/
```
Comment (adamancer) [inline CONTRIBUTING.md:116]: And this one

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -101,20 +101,20 @@ community listed at <https://carpentries.org/connect/> including via social
...
+[swc-issues]: https://github.com/issues?q=user%3Aswcarpentry/
```
Comment (adamancer) [inline CONTRIBUTING.md:117]: And this one

**Review by adamancer — CHANGES_REQUESTED** [review]
The trailing slash breaks the three issues links, other than that this looks good. Thanks!

**Comment by jd-foster** [comment]
🤦 
@adamancer Thanks for picking this up. Not sure why I was thinking I needed to add them to everything.

**Pull request merged**